### PR TITLE
Immutable Restricted Storage

### LINKED ISSUE #295: No way to obain a `RestrictStorage` for `&Storage`
The `restrict` method takes `&mut self`, even though there should be an immutable variant.

cc @Aceeri

### PR DESCRIPTION
**Pull request by Aceeri**
fixes #295

---
This change is [Reviewable](https://reviewable.io/reviews/slide-rs/specs/306)

### CODE REVIEW

File: src/storage/restrict.rs [hunk 1]
```diff
@@ -177,6 +186,38 @@ where
...
+            data: &self.data.inner,
+            entities: &self.entities,
+            phantom: PhantomData,
+        }
+    }
+
+    /// Builds an immutable, parallel `RestrictedStorage`.
+    pub fn par_restrict<'rf>(
+        &'rf self,
+    ) -> RestrictedStorage<'rf, 'st, &BitSet, T, &T::Storage, ParallelRestriction> {
```
Comment (torkleyy) [inline src/storage/restrict.rs]: Do we really need two methods/variants here?

Comment (Aceeri) [reply src/storage/restrict.rs]: Now that I think about it, probably not.

**Comment by torkleyy** [comment]
Thanks, looks good to me. r? @Xaeroxe @WaDelma

---

Reviewed 2 of 3 files at r1, 1 of 1 files at r2.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/slide-rs/specs/306#-:-L0AN6FSYzmSWB8Ewk8o:b-3zdryi)*

**Comment by torkleyy** [comment]
bors delegate=@Aceeri

I think it's fine as is, please merge as soon as you get another approval.

**Comment by Aceeri** [comment]
Going to finish up the other pr later today.

bors r+

**Pull request merged**